backend/app/main.py

```diff
@@ -15,24 +17,44 @@ def render(self, content: typing.Any) -> bytes:
         return orjson.dumps(content)
 
 
-origins = [
-    "http://localhost:3000",
-    "localhost:3000"
-]
-
-app = FastAPI(default_response_class=ORJSONResponse,
-              docs_url="/docs",
-              redoc_url=None,
-              title="CPT-Dashboard API Documentation",
-              version="0.0.1",
-              contact={
-                "name": "OCP PerfScale Jedi",
-                "url": "https://redhat.enterprise.slack.com/archives/C05CDC19ZKJ",
-              },
-              license_info={
-                "name": "Apache 2.0",
-                "url": "https://www.apache.org/licenses/LICENSE-2.0",
-              })
+origins = ["http://localhost:3000", "localhost:3000"]
+
+app = FastAPI(
+    default_response_class=ORJSONResponse,
+    docs_url="/docs",
+    redoc_url=None,
+    title="CPT-Dashboard API Documentation",
+    version="0.0.1",
+    contact={
+        "name": "OCP PerfScale Jedi",
+        "url": "https://redhat.enterprise.slack.com/archives/C05CDC19ZKJ",
+    },
+    license_info={
+        "name": "Apache 2.0",
+        "url": "https://www.apache.org/licenses/LICENSE-2.0",
+    },
+)
+
+
+@app.middleware("http")
+async def report_exceptions(request: Request, call_next):
```

Comment:
I'm not sure why, but the traceback from unhandled exceptions was getting swallowed somewhere except for a header message that wasn't useful. I added this in order to debug at all.

Like the other ancillary moving pieces, it's a small chunk of code that never seemed worth trying to separate -- and which I needed in order to debug. I can delete it if necessary, but that seems wasteful. 😦

Reply:
If it's independent of the rest of your changes and can be used elsewhere to debug, then it seems like it should be kept, and given its own pr.

Reply:
I'm fine with it being kept in this pr. I just wanted to flag it since this pr got way too big.

Reply:
Yeah, the PR got big, but the vast bulk of that really is new stuff. If I were confident of quick review and merge, "ancillary" bits like the pandas/numpy version fixes would be great candidates, especially since the generated lock files are enormous. However since we can't deploy the real work without those fixes, I'm reluctant to deal with that extra overhead if it's not really necessary. Pragmatism vs idealism is always a fine and twisty line in the mud!